### DISCUSSION TITLE
Shouldn't ScalarType::parseValue throw UserError instead of Error?

### DISCUSSION THREAD

**Discussion by davidPaymefy** [webonyx/graphql-php#1043] [Q&A] [answered]
Hello, I have this problem:

When a client sends a string instead of an integer as input ("10" vs 10), IntType throws a GraphQL\Error\Error and pollutes the log. I don't want to log this errors. 

I need a way to distinguish errors due to user sending wrong input from other internal errors.  Shouldn't ScalarType::parseValue throw UserError instead of Error, or am I wrong?

I would appreciate some feedback on this. Thanks in advance.

**Comment by spawnia** [comment, marked as answer]
See https://github.com/webonyx/graphql-php/blob/7cd74120cba023c603b806034912daaaf59718db/src/Error/Error.php#L21-L22

> I don't want to log this errors.

Then don't.

**Reply by davidPaymefy** [reply]
Thanks for you reply, but what does `execute phase` mean? Could it be an internal error? Because, if yes, I do want to log that.

**Reply by spawnia** [reply]
It is a user-facing error.

**Reply by davidPaymefy** [reply]
That's not what I'm asking. As and example, making a parallelism with an API REST, both a 500 and a 404 response are user-facing errors, but I will only want to log the 500.

**Reply by spawnia** [reply]
My response was not clearly phrased. `GraphQL\Error\Error` is used for errors caused by the user. For developer errors, this library uses `GraphQL\Error\InvariantViolation`.

**Reply by davidPaymefy** [reply]
Ok, I see, thank you very much. Everything is clear now.
